### DISCUSSION TITLE
how to contribute to time series modules in Pycaret

### DISCUSSION THREAD

**Discussion by wangzhishi** [pycaret/pycaret#2462] [General]
Hello there, I'm a developer of the open-source time series package [orbit](https://github.com/uber/orbit). We are interested in contributing to the time series functionality in pycaret and make orbit be part of it. Some questions in general:

1. What is the best way for new developers who want to work on pycaret?
2. Which branch/folder should I look at for the time series module in pycaret?
3. It seems that pycaret is leveraging sktime for the time series related. Does it mean it makes more sense we should go with sktime first if we want to have orbit adopted in pycaret?

Looking forward to hearing back.

Thanks,

**Comment by ngupta23** [comment]
Hi @wangzhishi Thanks for getting in touch. 

Are you looking to just add `orbit` to the `pycaret` module or are looking to contribute beyond that with general purpose time series capability development in `pycaret` time series?

If you are looking to help with general purpose development in `pycaret`, let me know and I will be happy to have a call to discuss.

If you are looking to add `orbit` only, then as you mentioned, this would need to be supported by `sktime` first. Once that is done, you can include this in pycaret with relative ease (using a model container), e.g. https://github.com/pycaret/pycaret/blob/85f795e0ac1a86984a90136ecd4dfafa6711a684/pycaret/containers/models/time_series.py#L1175

Regarding your other questions. 

> Which branch/folder should I look at for the time series module in pycaret?

We are in the process of releasing 3.0.0 version of `pycaret`. I would recommend waiting for this (maybe < 2-3 weeks) before making the contribution. I would think that it would take at least that long to add `orbit` to `sktime` so we should be good. 

We are also contemplating adding a `develop` branch with the 3.0.0 release (https://github.com/pycaret/pycaret/issues/2374). If we go ahead with this plan, then any future contributions would need to go into `develop`.

**Reply by feyzaozen** [reply]
I want to contribute pycaret  time series module. But it was different from non-ts modules. I could not add additional functionalities. Can you help me?

**Comment by wangzhishi** [comment]
Hi @ngupta23, thanks for getting back so quickly.

At this moment, I would say I'd like to prioritize making `pycaret` support `orbit`. With that said, I will go with `sktime` first, then include it as you suggested above. Thank you,
